pkgs/by-name/ft/ftb-app/package.nix

```
@@ -0,0 +1,43 @@
+{
+  lib,
+  fetchurl,
+  appimageTools,
+  makeDesktopItem,
+}: let
+  pname = "ftb-app";
+  version = "1.27.2";
+
+  src = fetchurl {
+    url = "https://piston.feed-the-beast.com/app/ftb-app-linux-${version}-x86_64.AppImage";
+    hash = "sha256-35GEI1OBvVkUvHvQAzzGz8ux9h+5W3acH0Wr5VkqyBw=";
+  };
+  
+  appimageContents = appimageTools.extractType2 { inherit pname src version; };
+in
+  appimageTools.wrapType2 {
+    inherit pname src version;
+    
+    extraInstallCommands = ''
+      for size in 16x16 32x32 48x48 64x64 128x128 256x256 512x512; do
+        install -Dm644 ${appimageContents}/usr/share/icons/hicolor/$size/apps/ftb-app.png \
+          $out/share/icons/hicolor/$size/apps/ftb-app.png
+      done
+      
+      install -Dm644 ${appimageContents}/ftb-app.desktop \
+        $out/share/applications/ftb-app.desktop
+      substituteInPlace $out/share/applications/ftb-app.desktop \
+        --replace-fail 'Exec=AppRun' 'Exec=${placeholder "out"}/bin/ftb-app'
+    '';
+
+    passthru.updateScript = nix-update-script { };
+    
+    meta = {
+      description = "Feed the Beast desktop app";
+      homepage = "https://www.feed-the-beast.com/ftb-app";
+      changelog = "https://www.feed-the-beast.com/ftb-app/changes#${version}";
+      license = lib.licenses.lgpl21Only;
+      maintainers = with lib.maintainers; [ nagymathev ];
+      mainProgram = "ftb-app";
+      platforms = lib.platforms.linux;
```

```suggestion
      platforms = [ "x86_64-linux" ];
```
(Based on the file name of src)

---

It also supports ARM, but it will need to add it as a source too

---

How would you go about supporting ARM too?

---

> How would you go about supporting ARM too?

I will give you a review soon, suggesting changes for adding support for arm linux and also darwin